indico/modules/rb_new/client/js/components/RoomListFilters.jsx

```diff
@@ -0,0 +1,185 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import propTypes from 'prop-types';
+import React from 'react';
+import {Button, Form, Icon, Input, Popup, Select} from 'semantic-ui-react';
+import {DebounceInput} from 'react-debounce-input';
+
+import {Param, Translate} from 'indico/react/i18n';
+import {parseRoomListFiltersText} from '../util';
+
+import './RoomListFilters.module.scss';
+
+
+export default class RoomListFilters extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            filtersVisible: false,
+            text: '',
+            building: '',
+            floor: ''
+        };
+    }
+
+    componentDidMount() {
+        const {fetchBuildings} = this.props;
+        fetchBuildings();
+    }
+
+    handleVisibleChange(visible) {
+        this.setState({filtersVisible: visible});
+    }
+
+    applyFilters() {
+        const {onConfirm} = this.props;
+        this.setState({filtersVisible: false});
+        onConfirm();
+    }
+
+    handleFiltersChange(filter, value) {
```

Comment:
`updateFilter`?

Reply:
The function looks a bit convoluted. What about separating it into `updateTextFilter(text)` and `updateFilter(filter, value)`.
Parsing the string is only be needed in `updateTextFilter`, and updating the text  only in `updateFilter`.